autogen/safer-cluster/variables.tf.tmpl

```diff
@@ -303,3 +303,9 @@ variable "enable_shielded_nodes" {
   default     = true
 }
 
+variable "enable_private_endpoint" {
+  type        = bool
+  description = "(Beta) Whether the master's internal IP address is used as the cluster endpoint"
+  default     = false
```

Comment:
This has the opposite effect of what its name implies. In order to follow the GKE hardening guides recommendation at https://cloud.google.com/kubernetes-engine/docs/how-to/hardening-your-cluster#restrict_network_access_to_the_control_plane_and_nodes ("Public endpoint access enabled, master authorized networks enabled"), this `enable_private_endpoint` needs to be set to `false`.


See also https://stackoverflow.com/a/58651940/6720110